File: Andriod/Constructor/construction/src/main/java/com/palitri/openiot/construction/framework/web/models/configurations/project/ProjectConfiguration.java
package com.palitri.openiot.construction.framework.web.models.configurations.project;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;

public class ProjectConfiguration implements Serializable {
    public ArrayList<Peripheral> peripherals;
    public ArrayList<PeripheralProperty> properties;

    public ProjectConfiguration()
    {
        this.Clear();
    }

    public void Clear()
    {
        this.peripherals = new ArrayList<Peripheral>();
        this.properties = new ArrayList<PeripheralProperty>();
    }

    public Peripheral GetPropertyPeripheral(int semantic)
    {
        for (Peripheral peripheral : this.peripherals)
            for (PeripheralProperty property : peripheral.properties)
                if (property.semantic == semantic)
                    return peripheral;

        return null;
    }

    public boolean LoadFromJSONString(String jsonString)
    {
        try {
            return this.LoadFromJSON(jsonString == null ? null : new JSONObject(jsonString));
        } catch (JSONException e) {
            e.printStackTrace();
            return false;
        }
    }

    public boolean LoadFromJSON(JSONObject jsonObject)
    {
        this.peripherals.clear();
        this.properties.clear();

        if (jsonObject == null)
            return false;

        try {
            if (jsonObject.has("peripherals"))
            {
                JSONArray jsonPeripherals = jsonObject.getJSONArray("peripherals");

                for (int propIndex = 0; propIndex < jsonPeripherals.length(); propIndex++)
                {
                    JSONObject jsonPeripheral = jsonPeripherals.getJSONObject(propIndex);

                    Peripheral per = new Peripheral();
                    per.LoadFromJSON(jsonPeripheral);

                    this.peripherals.add(per);

                    for (PeripheralProperty prop : per.properties)
                        this.properties.add(prop);
                }
            }

        } catch (JSONException e) {
            e.printStackTrace();
            return false;
        }

        return true;
    }

}

File: Windows/OpenIoT.App/OpenIoT.Lib/Web/Models/Configurations/Project/ProjectConfiguration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenIoT.Lib.Web.Models.Configurations.Project
{
    public class ProjectConfiguration
    {
        public List<Peripheral> Peripherals { get; set; }

        private List<PeripheralProperty> _properties = null;
        public List<PeripheralProperty> Properties 
        { 
            get
            {
                if (this._properties == null)
                    this._properties = this.Peripherals.SelectMany(p => p.Properties).ToList();

                return this._properties;
            }
        }

        public ProjectConfiguration()
        {
            this.Peripherals = new List<Peripheral>();
            this._properties = null;
        }

        public void Clear()
        {
            this.Peripherals.Clear();
            this._properties = null;
        }

        public Peripheral GetPropertyPeripheral(int semantic)
        {
            foreach (Peripheral peripheral in this.Peripherals)
                foreach (PeripheralProperty property in peripheral.Properties)
                    if (property.Semantic == semantic)
                        return peripheral;

            return null;
        }
    }
}
